src/graphforge/executor/evaluator.py

```
@@ -2858,6 +2858,12 @@ def _evaluate_temporal_function(func_name: str, args: list[CypherValue]) -> Cyph
                     dt = dt.replace(tzinfo=datetime.timezone.utc)
 
                 return CypherDateTime(dt)
+            elif isinstance(arg, CypherDateTime):
+                # datetime(datetime) → return as-is
+                # datetime(localdatetime) → add UTC timezone
+                if arg.value.tzinfo is not None:
+                    return arg
+                return CypherDateTime(arg.value.replace(tzinfo=datetime.timezone.utc))
             else:
                 raise TypeError(f"DATETIME expects string or map, got {type(arg).__name__}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)